bench/benchmark_cpdist.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import importlib
+import random
+import string
+from timeit import timeit
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+
+random.seed(18)
+
+plt.rc("font", size=13)  # controls default text sizes
+plt.rc("axes", titlesize=18)  # fontsize of the axes title
+plt.rc("axes", labelsize=15)  # fontsize of the x and y labels
+plt.rc("xtick", labelsize=15)  # fontsize of the tick labels
+plt.rc("ytick", labelsize=15)  # fontsize of the tick labels
+plt.rc("legend", fontsize=15)  # legend fontsize
+
+LIBRARIES = (
+    "ratio",
+    "partial_ratio",
+    "token_sort_ratio",
+    "token_set_ratio",
+    "partial_token_sort_ratio",
+    "partial_token_set_ratio",
+    "QRatio",
+    "WRatio",
+)
+
+
+def load_func(target):
+    modname, funcname = target.rsplit(".", maxsplit=1)
+
+    module = importlib.import_module(modname)
+    return getattr(module, funcname)
+
+
+def get_platform():
+    import platform
+
+    uname = platform.uname()
+    pyver = platform.python_version()
+    return f"Python {pyver} on {uname.system} ({uname.machine})"
+
+
+def benchmark():
+    words = ["".join(random.choice(string.ascii_letters + string.digits) for _ in range(8)) for _ in range(1000000)]
+    sample_rate = len(words) // 2
+    words1 = words[:sample_rate]
+    words2 = words[sample_rate::]
+    total = len(words1) + len(words2)
```

Comment:
This is the total count of comparisons. So this should only be
```suggestion
    total = len(words1)
```